daq_macros.py

```diff
@@ -1867,17 +1871,23 @@ def snakeRasterBluesky(rasterReqID, grain=""):
     if govStatus.exception():
       logger.error(f"Problem during start-of-raster governor move, aborting! exception: {govStatus.exception()}")
       return
-    raster_flyer.configure_detector(file_prefix=rasterFilePrefix, data_directory_name=data_directory_name)
-    raster_flyer.detector.stage()
+    #raster_flyer.configure_detector(file_prefix=rasterFilePrefix, data_directory_name=data_directory_name)
+    #raster_flyer.detector.stage()
     procFlag = int(getBlConfig("rasterProcessFlag"))
     spotFindThreadList = []
+    yield from bps.mv(samplexyz.omega, (omega-1)) # attempting to over-compensate omega movement
     for row_index, row in enumerate(rows):  # since we have vectors in rastering, don't move between each row
-        xMotAbsoluteMove, xEnd, yMotAbsoluteMove, yEnd, zMotAbsoluteMove, zEnd = raster_positions(row, stepsize, omegaRad, rasterStartX, rasterStartY, rasterStartZ, row_index)
-        vector = {'x': (xMotAbsoluteMove, xEnd), 'y': (yMotAbsoluteMove, yEnd), 'z': (zMotAbsoluteMove, zEnd)}
+        logger.info(f'starting new row: {row_index}')
+        zMotAbsoluteMove, zEnd, yMotAbsoluteMove, yEnd, xMotAbsoluteMove, xEnd = raster_positions(row, stepsize, omegaRad+90, rasterStartZ*1000, rasterStartY*1000, rasterStartX*1000, row_index)
+        vector = {'x': (xMotAbsoluteMove/1000, xEnd/1000), 'y': (yMotAbsoluteMove/1000, yEnd/1000), 'z': (zMotAbsoluteMove/1000, zEnd/1000)}
+        yield from bps.mv(samplexyz.x, xMotAbsoluteMove/1000)
+        yield from bps.mv(samplexyz.y, yMotAbsoluteMove/1000)
+        yield from bps.mv(samplexyz.z, zMotAbsoluteMove/1000)
+        yield from bps.mv(samplexyz.omega, omega-0.05)
         yield from zebraDaqRasterBluesky(raster_flyer, omega, numsteps, img_width_per_cell * numsteps, img_width_per_cell, exptimePerCell, rasterFilePrefix,
             data_directory_name, file_number_start, row_index, vector)
-        raster_flyer.zebra.reset.put(1)  # reset after every row to make sure it is clear for the next row
-        time.sleep(0.2)  # necessary for reliable row processing - see comment in commit 6793f4
+        #raster_flyer.zebra.reset.put(1)  # reset after every row to make sure it is clear for the next row
+        time.sleep(0.3)  # necessary for reliable row processing - see comment in commit 6793f4
```

Comment:
use `yield from bps.sleep(...)` in plans.  Putting time.sleep will block the event loop in bad ways.

Reply:
fixed with [d706a4b](https://github.com/NSLS-II/lsdc/pull/314/commits/d706a4b190cc4c4a7169dfeaaa553272534189a8)